### PR TITLE
Move WithContext into the resource admission controller.

### PR DESCRIPTION
**Pull request by mattmoor**
WIP until https://github.com/knative/pkg/pull/819 lands

### CODE REVIEW

**Comment by vagababov** [comment]
/lgtm
/hold 
on the base pr.

**Comment by mattmoor** [comment]
/hold cancel

That's what WIP is for 🤣

File: webhook/resource_admission_controller.go [hunk 1]
```diff
@@ -59,18 +59,22 @@ type ResourceAdmissionController struct {
 	handlers map[schema.GroupVersionKind]GenericCRD
 
 	disallowUnknownFields bool
+	WithContext           func(context.Context) context.Context
```
Comment (vagababov) [inline webhook/resource_admission_controller.go]: does this need to be public?

Comment (mattmoor) [reply webhook/resource_admission_controller.go]: Not currently, but when this moves to its own package, the test update I made will need this to be public 🙄

Comment (mattmoor) [reply webhook/resource_admission_controller.go]: added a comment

**Pull request merged**